gobblin-api/src/main/java/org/apache/gobblin/publisher/DataPublisher.java

```diff
@@ -30,7 +33,16 @@
 /**
  * Defines how to publish data and its corresponding metadata. Can be used for either task level or job level publishing.
  */
-public abstract class DataPublisher implements Closeable {
+public abstract class DataPublisher implements Closeable, CapabilityAware {
+  /**
+   * Threadsafe capability.
+   */
+  public static final Capability THREADSAFE = new Capability("THREADSAFE", false);
```

Comment:
Should this be part of `DataPublisher` or of a common set of capabilities?

Reply:
Moved `THREADSAFE` to the `Capability` class. Leaving REUSABLE here since that is specific to `DataPublisher`s.